### PR TITLE
refactor inputs

### PR DESCRIPTION
**Pull request by ianmacartney**
Goals:

- Co-locate inputs and code that uses it: in particular agent-specific inputs & handling
- Simplify the input name → handleXX lookup.
- For agents you can stay within one file when jumping between the code that writes and handles inputs.
- Reduce boilerplate for adding an input
- Co-locate the input args and implementation.

### CODE REVIEW

File: convex/engine/game.ts [hunk 1]
```diff
@@ -1,18 +1,23 @@
...
+//   Args extends PropertyValidators,
+//   ReturnValue extends any,
+//   G extends any,
+// > = {
+//   args: Args;
+//   handler: (game: G, now: number, args: ObjectType<Args>) => Promise<ReturnValue>;
+// };
 
-export type InputHandlers = Record<string, InputHandler<any, any>>;
+// export type InputHandlers = Record<string, InputHandler<any, any, any>>;
```
Comment (sujayakar) [inline convex/engine/game.ts:16]: let's just remove for now then? (and below)

File: convex/game/inputs.ts [hunk 2]
```diff
@@ -1,425 +1,240 @@
...
   },
-
-  // Move the player to a specified location.
-  moveTo: {
-    args: v.object({
-      playerId: v.id('players'),
-      destination: v.union(point, v.null()),
-    }),
-    returnValue: v.null(),
+} as const);
```
Comment (sujayakar) [inline convex/game/inputs.ts:73]: ah interesting, why do we need the `as const` here but not in the agent handlers?

File: convex/game/inputs.ts [hunk 3]
```diff
@@ -1,425 +1,240 @@
...
-    args: v.object({
-      playerId: v.id('players'),
-      conversationId: v.id('conversations'),
-    }),
-    returnValue: v.null(),
+});
+// Start a conversation, inviting the specified player.
+// Conversations can only have two participants for now,
+// so we don't have a separate "invite" input.
+const startConversation = inputHandler({
```
Comment (sujayakar) [inline convex/game/inputs.ts:108]: might even be nice to move some of these to `conversations.ts` to match what we're doing with the agents

**Review by sujayakar — APPROVED** [review]
much cleaner!

File: convex/engine/game.ts (hunk 1 above)
Comment (ianmacartney) [reply convex/engine/game.ts:16]: sounds good. wanted to keep it around while you checked it out to see if you wanted to try to add something here short-term. we can find it in the git tree if we need to

File: convex/game/inputs.ts (hunk 2 above)
Comment (ianmacartney) [reply convex/game/inputs.ts:73]: good catch- this was leftover from trying to get codegen to associate the name and args. It's currently working without this, so I'll remove it

**Pull request merged**